src/lua/api-gateway/tracking/log/tracingRulesLogger.lua

```diff
@@ -0,0 +1,88 @@
+--[[
+  Copyright 2016 Adobe Systems Incorporated. All rights reserved.
+
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under the License
+   is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR RESPRESENTATIONS OF ANY KIND,
+   either express or implied.  See the License for the specific language governing permissions and
+   limitations under the License.
+  ]]
+
+local cjson = require "cjson"
+local su = require "api-gateway.tracking.util.stringutil"
+
+local _M = {}
+
+function _M:new(o)
+    local o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    return o
+end
+
+function _M:buildTraceMessage(rule)
+    local data = {}
+
+    data["id"] = tostring(rule.id)
+    data["domain"] = tostring(rule.domain)
+    data["request_body"] = tostring(ngx.var.request_body)
+    data["status"] = tostring(ngx.var.status)
+
+    for key,value in pairs(ngx.header) do
```

Comment:
Indeed, we should be using maybe [`ngx.req.get_headers`](https://github.com/openresty/lua-nginx-module#ngxreqget_headers) instead.